File: android/src/main/java/com/zhangguoliang/amap_plugin/AmapPlugin.java
package com.zhangguoliang.amap_plugin;

import io.flutter.plugin.common.PluginRegistry.Registrar;

/** AmapPlugin */
public class AmapPlugin {
  private static final String _viewType = "flutter_native_mapview";

  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    registrar.platformViewRegistry().registerViewFactory(_viewType, new AMapViewFactory(registrar.messenger()));
  }

}

File: ios/Classes/AmapPlugin.m
#import "AmapPlugin.h"
#import "AMapViewFactory.h"
@implementation AmapPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    static NSString *identifier = @"flutter_native_mapview";
    [registrar registerViewFactory:[[AMapViewFactory alloc] initWithMessenger:registrar.messenger] withId:identifier];
}


@end
